UnitsNet/UnitConverter.cs

```diff
@@ -395,22 +397,30 @@ private static bool TryParseUnit(Type unitType, string unitName, out object unit
             return true;
         }
 
+        private static List<Type> UnitTypes = UnitsNetAssembly.GetTypes()
+            .Where(x => x.FullName.StartsWith(UnitTypeNamespace))
+            .ToList();
+
         private static bool TryGetUnitType(string quantityName, out Type unitType)
         {
-            string unitTypeName = $"{UnitTypeNamespace}.{quantityName}Unit";
+            quantityName += "Unit";
+            unitType = QuantityTypes.FirstOrDefault(x => 
+                x.Name.Equals(quantityName, StringComparison.OrdinalIgnoreCase));
 
-            unitType = UnitsNetAssembly.GetType(unitTypeName); // ex: UnitsNet.Units.LengthUnit enum
             if(unitType == null)
                 return false;
 
             return true;
         }
 
+        private static List<Type> QuantityTypes = UnitsNetAssembly.GetTypes()
```

Comment:
Make it `readonly` and of type `IReadOnlyList<Type>`. Same for `UnitTypes` above. 
Move all fields to the top, above any methods or constructors.